src/controllers/ajax/IndexAjaxController.php

```diff
@@ -119,6 +131,25 @@ protected function getActions(): array<string> {
 
     // Check logo
     $final_logo = $logo;
+
+    if ($is_custom_logo) {
+      $logo_data = str_replace(' ', '+', $final_logo);
+      $logo_data_unbase64ed = base64_decode($logo_data);
+
+      $name = 'custom-'.time();
+      $filename = $name.'.'.$logo_type;
+      // '..' is separated for DB storage
+      $url = self::CUSTOM_LOGO_DIR.$filename;
+      $filepath = dirname(__FILE__).'/../..'.$url;
+      file_put_contents($filepath, $logo_data_unbase64ed);
+
+      // Check validity of logo file
+      list($width, $height, $type, $attr) = getimagesize($full_filename);
```

Comment:
This check should be done before putting the file in the final destination. Also here is where you want to make sure the uploaded file is not a ```.php``` file. Easiest way would be to limit the extensions to png, gif and jpg.